AppCenterAnalytics/AppCenterAnalytics/Internals/Model/MSEventLog.m

```diff
@@ -98,11 +98,46 @@ - (void)setPropertiesAndMetadataForCSLog:(MSCommonSchemaLog *)csLog {
   if (self.typedProperties) {
     csProperties = [NSMutableDictionary new];
     metadata = [NSMutableDictionary new];
+    NSString *baseDataPrefix = [NSString stringWithFormat:@"%@.", kMSDataBaseData];
+
+    // If there is no valid "baseType" property, there must not be "baseData.*" properties.
+    if (![self.typedProperties.properties[kMSDataBaseType] isKindOfClass:[MSStringTypedProperty class]]) {
```

Comment:
nit: can you swap if block and else block by removing `!` from the condition?
```
if (!condition) {
  // Do A
} else {
  // Do B
}
```
can be
```
if (condition) {
  // Do B
} else {
  // Do A
}
```